tests/bluetooth/tester/src/btp_gap.c

```diff
@@ -45,38 +45,28 @@ static uint8_t oob_legacy_tk[16] = { 0 };
 
 static bool filter_list_in_use;
 
-#if !defined(CONFIG_BT_SMP_OOB_LEGACY_PAIR_ONLY)
 static struct bt_le_oob oob_sc_local = { 0 };
 static struct bt_le_oob oob_sc_remote = { 0 };
-#endif /* !defined(CONFIG_BT_SMP_OOB_LEGACY_PAIR_ONLY) */
 
 /* connection parameters for rejection test */
 #define REJECT_INTERVAL_MIN 0x0C80
 #define REJECT_INTERVAL_MAX 0x0C80
 #define REJECT_LATENCY 0x0000
 #define REJECT_SUPERVISION_TIMEOUT 0x0C80
 
-#if defined(CONFIG_BT_PRIVACY)
 static struct {
 	bt_addr_le_t addr;
 	bool supported;
 } cars[CONFIG_BT_MAX_PAIRED];
 
-static uint8_t read_car_cb(struct bt_conn *conn, uint8_t err,
-			  struct bt_gatt_read_params *params, const void *data,
-			  uint16_t length);
-
-static struct bt_gatt_read_params read_car_params = {
-		.func = read_car_cb,
-		.by_uuid.uuid = BT_UUID_CENTRAL_ADDR_RES,
-		.by_uuid.start_handle = BT_ATT_FIRST_ATTRIBUTE_HANDLE,
-		.by_uuid.end_handle = BT_ATT_LAST_ATTRIBUTE_HANDLE,
-};
-
 static uint8_t read_car_cb(struct bt_conn *conn, uint8_t err,
 			  struct bt_gatt_read_params *params, const void *data,
 			  uint16_t length)
 {
+	if (IS_ENABLED(CONFIG_BT_PRIVACY)) {
+		return -ENOSYS;
+	}
```

Comment:
This doesn't make any sense :) It's a callback that returns a `uint8_t`

Should be `BT_GATT_ITER_STOP` if anything, but why is this being guarded at all? The callback does not seem related to privacy.

Reply:
I've checked git history, it was originally guarded to satisfy GAP/CONN/DCON/BV-05-C, it exercises privacy mode behavior, in d80e480f2005 it describes that:
> This allows to do directed advertising with peer address set to RPA.
> To do this according to specification IUT must first read Central
> Address Resolution characteristic to check if peer supports it.
I guess it is possible to return BT_GATT_ITER_STOP, but how one would see that there was an error?

Reply:
> I've checked git history, it was originally guarded to satisfy GAP/CONN/DCON/BV-05-C, it exercises privacy mode behavior, in [d80e480](https://github.com/zephyrproject-rtos/zephyr/commit/d80e480f20050171d199fea79a1959bdbb75932d) it describes that:
> 
> > This allows to do directed advertising with peer address set to RPA.
> > To do this according to specification IUT must first read Central
> > Address Resolution characteristic to check if peer supports it.
> 
> I guess it is possible to return BT_GATT_ITER_STOP, but how one would see that there was an error?

You already guard the call that uses this callback, so if you want any check here, we might just add an assert. We could also just remove the check in the callback, as it should never happen

Reply:
@kedMertens This still need to be changed :)

Either remove the check (my preference), add an ASSERT or return `BT_GATT_ITER_STOP` with a log statement